src/server.c

```diff
@@ -127,9 +127,30 @@ void formatTimezone(char *buf, size_t buflen, int timezone, int daylight_active)
     buf[6] = '\0';
 }
 
+/* Modifies the input string by:
+ *      removing \r and \n
+ *      replacing " with '
+ * 
+ * Parameters:
+ *   msg - A string which will be modified in place. */
+void filterInvalidLogfmtChar(char *msg) {
+    if (msg == NULL) return;
+
+    int src = 0, dst = 0;
+    while (msg[src] != '\0') {
+        if (msg[src] == '"') {
+            msg[dst++] = '\'';
+        } else if (msg[src] != '\n' && msg[src] != '\r') {
+            msg[dst++] = msg[src];
+        }
+        src++;
```

Comment:
Just an idea: If we replace `\r` and `\n` with a space instead, then we don't need to copy/move the rest of the string. We don't need the `dst` index.

```suggestion
    while (msg[src] != '\0') {
        if (msg[src] == '"') {
            msg[src] = '\'';
        } else if (msg[src] == '\n' && msg[src] == '\r') {
            msg[src] = ' ';
        }
        src++;
```